### PR TITLE
AS3 declaration POST call implementation

### PR DESCRIPTION
**Pull request by agupta-F5**
This code will POST AS3 declaration to Big-IP.

1. Implemented function postAS3Declaration to POST AS3 declaration to Big IP.
2. Created new generic function restCallToBigIP to handle all REST calls.
3. Created new file as3Manager_test.go to write unit test cases.

Known Issue:
Needs to fetch Big IP URL and basic auth credential for REST calls to BigIP.

### CODE REVIEW

File: pkg/appmanager/as3Manager.go [hunk 1]
```diff
@@ -123,4 +139,53 @@ func buildAS3Declaration(obj as3Object, template as3Template) as3Declaration {
 //Story 5
 // Takes AS3 Declaration and posting it to BigIP
```
Comment (subbuv26) [inline pkg/appmanager/as3Manager.go]: Code comments need to be updated. 
Remove //Strory5.

File: pkg/appmanager/as3Manager.go [hunk 2]
```diff
@@ -123,4 +139,53 @@ func buildAS3Declaration(obj as3Object, template as3Template) as3Declaration {
...
+	}
+	req, err := http.NewRequest(method, as3RestClient.baseURL+route, data)
+	if err != nil {
+		log.Errorf("[as3_log] Creating new HTTP request error: %s ", err)
+		return string(body), false
+	}
+	req.SetBasicAuth(BigIPUsername, BigIPPassword)
+	resp, err := as3RestClient.Client.Do(req)
+	if err != nil {
+		log.Errorf("[as3_log] REST call error: %s ", err)
```
Comment (subbuv26) [inline pkg/appmanager/as3Manager.go]: Format specifier needs to be changed to %v

File: pkg/appmanager/as3Manager.go [hunk 3]
```diff
@@ -123,4 +139,53 @@ func buildAS3Declaration(obj as3Object, template as3Template) as3Declaration {
...
+	if status {
+		log.Infof("[as3_log] AS3 declaration POST to BIG-IP success")
+	}
+
+}
+
+// Takes AS3 Declaration and REST method and post it to BigIP
+func (as3RestClient *As3RestClient) restCallToBigIP(method string, route string, declaration as3Declaration) (string, bool) {
+	log.Debugf("[as3_log] REST call with AS3 Manager")
+	timeout := time.Duration(5 * time.Second)
```
Comment (ssurenr) [inline pkg/appmanager/as3Manager.go]: Let's increase this to 15 sec

File: pkg/appmanager/as3Manager.go [hunk 4]
```diff
@@ -123,4 +139,53 @@ func buildAS3Declaration(obj as3Object, template as3Template) as3Declaration {
...
+	var body []byte
+	//tr flag is set true to disable SSL validation
```
Comment (ssurenr) [inline pkg/appmanager/as3Manager.go]:
```suggestion
	//FIXME: tr flag is set to true to disable SSL validation
```

File: pkg/appmanager/as3Manager.go [hunk 5]
```diff
@@ -123,4 +139,53 @@ func buildAS3Declaration(obj as3Object, template as3Template) as3Declaration {
...
+		log.Errorf("[as3_log] REST call error: %s ", err)
+		return string(body), false
+	}
+	body, err = ioutil.ReadAll(resp.Body)
+	if err != nil {
+		log.Errorf("[as3_log] REST call error: %s ", err)
+		return string(body), false
+	}
+	log.Infof("[as3_log] AS3 REST call response: %s ", string(body))
+	defer resp.Body.Close()
```
Comment (ssurenr) [inline pkg/appmanager/as3Manager.go]: Please handle the error.

File: pkg/appmanager/as3Manager_test.go [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+	It("AS3 declaratyion POST call test", func() {
+		route := "/mgmt/shared/appsvcs/declare"
+		method := "POST"
+		var template as3Declaration = `{"class":"AS3","action":"deploy","persist":true,}`
+		server := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+			// Test request parameters
+			// equals(t, req.URL.String(), "/mgmt/shared/appsvcs/info")
+			// Send response to be tested
+			rw.Write([]byte("OK"))
```
Comment (ssurenr) [inline pkg/appmanager/as3Manager_test.go]: Error needs to be handled.

**Pull request closed without merging**